### PR TITLE
Update scrollbar logic

### LINKED ISSUE #2443: Adventure screen, position of the scrollbar cursor is not updated correctly for the list of heroes
In fheroes2, the position of the scrollbar cursor is not updated correctly for the list of heroes.

In the example, at the end of the action, the scrollbar cursor should be at the very top of the scrollbar.

In fheroes2:

https://user-images.githubusercontent.com/43583994/102706861-dbcae680-4295-11eb-841d-516f8def168c.mp4

### LINKED ISSUE #2444: Kingdom Overview screen, Heroes Overview, position of the scrollbar cursor is not updated correctly while using it
In fheroes2, the position of the scrollbar cursor is not updated correctly while using it.

In the example, at the end of the action, the scrollbar cursor should be at the very top of the scrollbar.

In fheroes2:

https://user-images.githubusercontent.com/43583994/102706889-28aebd00-4296-11eb-9c54-3b9d9b7a24a6.mp4

Similar issue: https://github.com/ihhub/fheroes2/issues/2443.

### LINKED ISSUE #2652: The scrollbar of marketplace should be divided into segments according to the quantity of maximum resources
The bar after dragging and releasing it should be sticked to the certain areas, where integer number of resources is shown.
Currently, if we drag the bar and stop it right before the place, where the resource number is raised, it doesn't add the resource till we reach the exact area.
In the example below, we're trading 5000k for one resource. But we should place the bar at the exact position in order to trade resource, (On a screenshots I'm releasing the mouse button, but still cannot trade any resource)
image
image
image
image
image
I suggest to divide the scrollbar area into segments according to maximum resource number able to trade. When we click right before the middle of this segment, bar should jump to the previous quantity area. If we click after the middle of the segment, bar should be sticked to the next resource quantity area.

### PR DESCRIPTION
**Pull request by ihhub**
Once mouse is released the position of the scrollbar should be updated as in the original game.

close #2652
close #2444
close #2443

### CODE REVIEW

**Comment by ihhub** [comment]
Hi @Branikolog , please test all scrollbars in the game with this change.

**Review by oleg-derevenetz — APPROVED** [review]
LGTM. However, clang-tidy is right about rounding :)

**Comment by Branikolog** [comment]
> Hi @Branikolog , please test all scrollbars in the game with this change.

Hi, @ihhub 
I've noticed, that when I'm starting to move with any hero both hero and town lists jump to it upper position which is really inconvenient.

File: src/fheroes2/gui/ui_scrollbar.cpp [hunk 1]
```diff
@@ -129,7 +142,9 @@ namespace fheroes2
             else if ( posX > maxXPos )
                 posX = maxXPos;
 
-            _currentIndex = ( posX - _area.x ) * ( _maxIndex - _minIndex ) / roiWidth;
+            const double tempPos = static_cast<double>( posX - _area.x ) * ( _maxIndex - _minIndex ) / roiWidth;
+            _currentIndex = static_cast<int>( tempPos + 0.5 ) + _minIndex;
```
[the remark this answers is not in the document]
Comment (oleg-derevenetz) [reply src/fheroes2/gui/ui_scrollbar.cpp:146]: Hi @ihhub I fixed issue with incorrect markdown (like in this clang-tidy warning) in [this](https://github.com/platisd/clang-tidy-pr-comments/pull/12) PR, it is already merged into the upstream, you may want to update the [fheroes2/clang-tidy-pr-comments](https://github.com/fheroes2/clang-tidy-pr-comments) to the actual version :)

[the remark this answers is not in the document]
Comment (ihhub) [reply src/fheroes2/gui/ui_scrollbar.cpp:146]: Done.

**Comment by Branikolog** [comment]
@ihhub 
Sorry, the issue I mentioned is not connected with your changes. Actually, there's an opened issue already: #3147

**Pull request merged**